src/cljs/netrunner/gameboard.cljs

```diff
@@ -943,121 +945,127 @@
     (remove (set restricted-servers) servers)))
 
 (defn button-pane [{:keys [side active-player run end-turn runner-phase-12 corp-phase-12 corp runner me opponent] :as cursor} owner]
-  (om/component
-   (sab/html
-    [:div.button-pane {:on-mouse-over card-preview-mouse-over
-                       :on-mouse-out  card-preview-mouse-out}
-     (if-let [prompt (first (:prompt me))]
-       [:div.panel.blue-shade
-        [:h4 (for [item (get-message-parts (:msg prompt))] (create-span item))]
-        (if-let [n (get-in prompt [:choices :number])]
-          [:div
-           [:div.credit-select
-            [:select#credit {:default-value (get-in prompt [:choices :default] 0)}
-             (for [i (range (inc n))]
-               [:option {:value i} i])]]
-           [:button {:on-click #(send-command "choice"
-                                              {:choice (-> "#credit" js/$ .val js/parseInt)})}
-            "OK"]]
-          (cond
-               ;; choice of number of credits
-            (= (:choices prompt) "credit")
-            [:div
-             [:div.credit-select
-              [:select#credit (for [i (range (inc (:credit me)))]
-                                [:option {:value i} i])] " credits"]
-             [:button {:on-click #(send-command "choice"
-                                                {:choice (-> "#credit" js/$ .val js/parseInt)})}
-              "OK"]]
-               ;; choice of specified counters on card
-            (:card-title (:choices prompt))
-            [:div
-             [:div.credit-select
-              [:input#card-title {:placeholder "Enter a card title"
-                                  :onKeyUp #(when (= 13 (.-keyCode %))
-                                              (-> "#card-submit" js/$ .click)
-                                              (.stopPropagation %))}]]
-             [:button#card-submit {:on-click #(send-command "choice"
-                                                            {:choice (-> "#card-title" js/$ .val)})}
-              "OK"]
-             (when-let [autocomp (:autocomplete (:choices prompt))]
-               (-> "#card-title" js/$ (.autocomplete (clj->js {"source" autocomp})))
-               nil)]
-            (:counter (:choices prompt))
-            (let [counter-type (keyword (:counter (:choices prompt)))
-                  num-counters (get-in prompt [:card :counter counter-type] 0)]
+  (reify
+    om/IDidUpdate
+    (did-update [this prev-props prev-state]
+      (when-let [autocomp (get-in cursor [side :prompt 0 :choices :autocomplete])]
+        (-> "#card-title" js/$ (.autocomplete (clj->js {"source" autocomp})))))
```

Comment:
More idiomatic would be to use a `ref` to `#card-title`, instead of finding it by id.